src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/scanner/ScannerImpl.kt

```diff
@@ -1959,48 +1934,36 @@ class ScannerImpl(
     }
 
     /**
-     * See the specification for details. `SnakeYAML` and `libyaml` allow tabs inside plain scalar
+     * See the specification for details. `SnakeYAML` and `libyaml` allow tabs inside plain scalar.
      */
     private fun scanPlainSpaces(): String {
-        var length = 0
-        while (reader.peek(length) == ' '.code || reader.peek(length) == '\t'.code) {
-            length++
-        }
+        val length = reader.peekCount { c -> c in " \t" }
         val whitespaces = reader.prefixForward(length)
         val lineBreak = scanLineBreak() ?: return whitespaces
 
         allowSimpleKey = true
-        var prefix = reader.prefix(3)
-        if ("---" == prefix || "..." == prefix && CharConstants.NULL_BL_T_LINEBR.has(reader.peek(3))) {
-            return ""
-        } else if (settings.parseComments && atEndOfPlain()) {
+        if (checkDocumentSeparator() || settings.parseComments && atEndOfPlain()) {
             return ""
-        } else {
-            val breaks = StringBuilder()
-            while (true) {
-                if (reader.peek() == ' '.code) {
-                    reader.forward()
-                } else {
-                    val lbOpt = scanLineBreak()
-                    if (lbOpt != null) {
-                        breaks.append(lbOpt)
-                        prefix = reader.prefix(3)
-                        if ("---" == prefix || "..." == prefix && CharConstants.NULL_BL_T_LINEBR.has(
-                                reader.peek(3)
-                            )
-                        ) {
-                            return ""
-                        }
-                    } else {
-                        break
-                    }
+        }
+
+        val breaks = StringBuilder()
+
+        while (true) {
+            reader.forwardWhile { c -> c in " \t" }
```

Comment:
Changing 'forward while char is a space' to 'forward while char is a space _or tab_' fixed several test suite tests.